### PR TITLE
React 16

### PR DESCRIPTION
**Pull request by ratneshraval**
#### What tests were added (if relevant), and do all tests (rspec, karma, and integration CALS tests) pass?
- 2 failing tests. Need to adjust them. 

#### Which User story does this relate to?
- https://www.pivotaltracker.com/story/show/151733886

#### Brief feature description
- Upgrade to react 16 and enzyme 3.

#### Do any commands needs to be run in order to run the project with this PR?
- bundle install, yarn install

### CODE REVIEW

File: test/javascript/rfa_forms/rfa01a_edit_view/phoneNumberCardsGroup_tests.js [hunk 1]
```diff
@@ -151,7 +151,10 @@ describe('Preferred logic', () => {
   it('allows to change number', () => {
     const newNumber = '8884442323'
     // why do I have to update this?
-    component.update()
+    spyOn(component.instance(), 'onPhoneFieldChange').and.callThrough()
+    component.setProps({
```
Comment (ratneshraval) [inline test/javascript/rfa_forms/rfa01a_edit_view/phoneNumberCardsGroup_tests.js]: Why do you have to setProps if you're doing simulate in the next line?

File: test/javascript/rfa_forms/rfa01a_edit_view/aboutApplicantCard_tests.js [hunk 2]
```diff
@@ -48,7 +48,7 @@ describe('Verify More About Applican', () => {
   })
   it('High Level Education DropDown Change', () => {
     // spyOn(aboutApplicantComp.instance().props.setParentState, setCardState).and.callThrough()
-    let higherEducationField = aboutApplicantComp.find('#highest_education_level')
+    let higherEducationField = aboutApplicantComp.find('#highest_education_level').hostNodes()
```
Comment (anuroop15) [inline test/javascript/rfa_forms/rfa01a_edit_view/aboutApplicantCard_tests.js]: why are we using hostNodes() for a mounted component. ?

Comment (ratneshraval) [reply test/javascript/rfa_forms/rfa01a_edit_view/aboutApplicantCard_tests.js]: Try running without hostNodes. It is needed.

File: test/javascript/search/search_tests.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 import SearchApp from '../../../app/javascript/search/search'
-var TestUtils = require('react-dom/lib/ReactTestUtils')
+var TestUtils = require('react-dom/test-utils')
```
Comment (Nasir2712) [inline test/javascript/search/search_tests.js]: Why are we not using shallow renderer in this file?

Comment (ratneshraval) [reply test/javascript/search/search_tests.js]: Not sure. I'm just adjusting existing tests to make it work with react 16.

File: test/javascript/rfa_forms/rfa01a_edit_view/phoneNumberCardsGroup_tests.js (hunk 1 above)
Comment (anuroop15) [reply test/javascript/rfa_forms/rfa01a_edit_view/phoneNumberCardsGroup_tests.js]: as we are changing props over-time, the component need to be re-rendered.  So using setProps.
ref: https://github.com/airbnb/enzyme/blob/master/docs/api/ShallowWrapper/setProps.md#setpropsnextprops--self

I have changed that as part of testing. will move spyOn method to beforeEach method.

File: test/javascript/rfa_forms/rfa01a_edit_view/aboutApplicantCard_tests.js (hunk 2 above)
Comment (anuroop15) [reply test/javascript/rfa_forms/rfa01a_edit_view/aboutApplicantCard_tests.js]: ok! got the usage of hostNodes().

**Pull request merged**